### DISCUSSION TITLE
phoebe 2.4.3  only surpport Kepler:mean, all errors when set as other passband

### DISCUSSION THREAD

**Discussion by dingxu6207** [phoebe-project/phoebe2#617] [Questions]
`The phoebe 2.4 only surpport band Kepler:mean
`b.add_dataset('lc', times=times, passband= 'Kepler:mean')

when I set passband= 'TESS:T', 
b.add_dataset('lc', times=times, passband= 'TESS:T')

---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Input In [2], in <cell line: 27>()
     24 b.flip_constraint('fillout_factor', solve_for='pot')
     25 b['fillout_factor'] = f    #0.61845703
---> 27 b.run_compute(irrad_method='none')

File ~/anaconda3/lib/python3.9/site-packages/phoebe/parameters/parameters.py:424, in send_if_client.<locals>._send_if_client(self, *args, **kwargs)
    421     return ret_
    423 else:
--> 424     return fctn(self, *args, **kwargs)

File ~/anaconda3/lib/python3.9/site-packages/phoebe/frontend/bundle.py:11913, in Bundle.run_compute(self, compute, model, solver, detach, dataset, times, return_changes, **kwargs)
  11908     self.compute_ld_coeffs(compute=compute, skip_checks=True, set_value=True, **{k:v for k,v in kwargs.items() if k in computeparams.qualifiers})
  11909     # NOTE that if pblum_method != 'phoebe', then system will be None
  11910     # otherwise the system will be create which we can pass on to the backend
  11911     # the phoebe backend can then skip initializing the system at least on the master proc
  11912     # (workers will need to recreate the mesh)
> 11913     system, pblums_abs, pblums_scale, pblums_rel, pbfluxes = self.compute_pblums(compute=compute, ret_structured_dicts=True, skip_checks=True, **{k:v for k,v in kwargs.items() if k in computeparams.qualifiers})
  11914     l3s = self.compute_l3s(compute=compute, use_pbfluxes=pbfluxes, ret_structured_dicts=True, skip_checks=True, skip_compute_ld_coeffs=True, **{k:v for k,v in kwargs.items() if k in computeparams.qualifiers})
  11915 else:

File ~/anaconda3/lib/python3.9/site-packages/phoebe/frontend/bundle.py:10695, in Bundle.compute_pblums(self, compute, model, pblum, pblum_abs, pblum_scale, pbflux, set_value, unit, **kwargs)
  10693     self.compute_ld_coeffs(compute=compute, set_value=True, skip_checks=True, **{k:v for k,v in kwargs.items() if k not in ['ret_structured_dicts', 'pblum_mode', 'pblum_method', 'skip_checks']})
  10694 # TODO: make sure this accepts all compute parameter overrides (distortion_method, etc)
> 10695 system = kwargs.get('system', self._compute_intrinsic_system_at_t0(compute=compute, datasets=pblum_datasets, atms=atms, **kwargs))
  10696 logger.debug("computing observables with ignore_effects=True for {}".format(pblum_datasets))
  10697 system.populate_observables(t0, ['lc'], pblum_datasets, ignore_effects=True)

File ~/anaconda3/lib/python3.9/site-packages/phoebe/frontend/bundle.py:10367, in Bundle._compute_intrinsic_system_at_t0(self, compute, datasets, compute_l3, compute_l3_frac, compute_extrinsic, **kwargs)
  10365 logger.debug("creating system with compute={} kwargs={}".format(system_compute, kwargs))
  10366 try:
> 10367     system = backends.PhoebeBackend()._compute_intrinsic_system_at_t0(self, system_compute, datasets=datasets, reset=False, lc_only=False, **kwargs)
  10368 except Exception as err:
  10369     restore_conf()

File ~/anaconda3/lib/python3.9/site-packages/phoebe/backend/backends.py:994, in PhoebeBackend._compute_intrinsic_system_at_t0(self, b, compute, dynamics_method, hier, meshablerefs, datasets, reset, lc_only, **kwargs)
    991 # kinds = [b.get_dataset(dataset=ds).kind for ds in datasets]
    993 system.update_positions(t0, x0, y0, z0, vx0, vy0, vz0, etheta0, elongan0, eincl0, ignore_effects=True)
--> 994 system.populate_observables(t0, ['lc' for dataset in datasets], datasets, ignore_effects=True)
    997 if reset:
    998     logger.debug("rank:{}/{} PhoebeBackend._create_system_and_compute_pblums: resetting system".format(mpi.myrank, mpi.nprocs))

File ~/anaconda3/lib/python3.9/site-packages/phoebe/backend/universe.py:319, in System.populate_observables(self, time, kinds, datasets, ignore_effects)
    317 for kind, dataset in zip(kinds, datasets):
    318     for starref, body in self.items():
--> 319         body.populate_observable(time, kind, dataset, ignore_effects=ignore_effects)

File ~/anaconda3/lib/python3.9/site-packages/phoebe/backend/universe.py:2993, in Envelope.populate_observable(self, time, kind, dataset, **kwargs)
   2988 """
   2989 TODO: add documentation
   2990 """
   2992 for half in self._halves:
-> 2993     half.populate_observable(time, kind, dataset, **kwargs)

File ~/anaconda3/lib/python3.9/site-packages/phoebe/backend/universe.py:1069, in Body.populate_observable(self, time, kind, dataset, ignore_effects, **kwargs)
   1061 if time==self.time and dataset in self.populated_at_time and 'pblum' not in kind:
   1062     # then we've already computed the needed columns
   1063 
   1064     # TODO: handle the case of intensities already computed by
   1065     # /different/ dataset (ie RVs computed first and filling intensities
   1066     # and then lc requesting intensities with SAME passband/atm)
   1067     return
-> 1069 new_mesh_cols = getattr(self, '_populate_{}'.format(kind.lower()))(dataset, ignore_effects=ignore_effects, **kwargs)
   1071 for key, col in new_mesh_cols.items():
   1073     self.mesh.update_columns_dict({'{}:{}'.format(key, dataset): col})

File ~/anaconda3/lib/python3.9/site-packages/phoebe/backend/universe.py:1795, in Star._populate_lc(self, dataset, ignore_effects, **kwargs)
   1791 pblum = kwargs.get('pblum', 4*np.pi)
   1793 if lc_method=='numerical':
-> 1795     pb = passbands.get_passband(passband)
   1797     if ldatm != 'none' and '{}:ldint'.format(ldatm) not in pb.content:
   1798         if ld_mode == 'lookup':

File ~/anaconda3/lib/python3.9/site-packages/phoebe/atmospheres/passbands.py:3239, in get_passband(passband, content, reload, update_if_necessary, download_local, download_gzipped)
   3237 if reload or _pbtable.get(passband, {}).get('pb', None) is None:
   3238     logger.info("loading {} passband from {} (including all tables)".format(passband, _pbtable[passband]['fname']))
-> 3239     pb = Passband.load(_pbtable[passband]['fname'], load_content=True)
   3240     _pbtable[passband]['pb'] = pb
   3242 return _pbtable[passband]['pb']

File ~/anaconda3/lib/python3.9/site-packages/phoebe/atmospheres/passbands.py:481, in Passband.load(cls, archive, load_content)
    479 if 'ck2004:Inorm' in self.content:
    480     self._ck2004_axes = (np.array(list(hdul['ck_teffs'].data['teff'])), np.array(list(hdul['ck_loggs'].data['logg'])), np.array(list(hdul['ck_abuns'].data['abun'])))
--> 481     self._ck2004_energy_grid = hdul['cknegrid'].data
    482     self._ck2004_photon_grid = hdul['cknpgrid'].data
    484 if 'ck2004:Imu' in self.content:

File ~/anaconda3/lib/python3.9/site-packages/astropy/utils/decorators.py:777, in lazyproperty.__get__(self, obj, owner)
    775         val = obj_dict.get(self._key, _NotFound)
    776         if val is _NotFound:
--> 777             val = self.fget(obj)
    778             obj_dict[self._key] = val
    779 return val

File ~/anaconda3/lib/python3.9/site-packages/astropy/io/fits/hdu/image.py:235, in _ImageBaseHDU.data(self)
    232 if len(self._axes) < 1:
    233     return
--> 235 data = self._get_scaled_image_data(self._data_offset, self.shape)
    236 self._update_header_scale_info(data.dtype)
    238 return data

File ~/anaconda3/lib/python3.9/site-packages/astropy/io/fits/hdu/image.py:779, in _ImageBaseHDU._get_scaled_image_data(self, offset, shape)
    771 """
    772 Internal function for reading image data from a file and apply scale
    773 factors to it.  Normally this is used for the entire image, but it
    774 supports alternate offset/shape for Section support.
    775 """
    777 code = BITPIX2DTYPE[self._orig_bitpix]
--> 779 raw_data = self._get_raw_data(shape, code, offset)
    780 raw_data.dtype = raw_data.dtype.newbyteorder('>')
    782 if self._do_not_scale_image_data or (
    783         self._orig_bzero == 0 and self._orig_bscale == 1 and
    784         self._blank is None):
    785     # No further conversion of the data is necessary

File ~/anaconda3/lib/python3.9/site-packages/astropy/io/fits/hdu/base.py:522, in _BaseHDU._get_raw_data(self, shape, code, offset)
    519     return np.ndarray(shape, dtype=code, buffer=self._buffer,
    520                       offset=offset)
    521 elif self._file:
--> 522     return self._file.readarray(offset=offset, dtype=code, shape=shape)
    523 else:
    524     return None

File ~/anaconda3/lib/python3.9/site-packages/astropy/io/fits/file.py:324, in _File.readarray(self, size, offset, dtype, shape)
    321             else:
    322                 raise
--> 324     return np.ndarray(shape=shape, dtype=dtype, offset=offset,
    325                       buffer=self._mmap)
    326 else:
    327     count = reduce(operator.mul, shape)

TypeError: buffer is too small for requested array

**Comment by aprsa** [comment]
I can't reproduce this with phoebe 2.4.3. Can you please do:
```
import phoebe
pb = phoebe.get_passband('TESS:T')
```
Does that work?

**Reply by dingxu6207** [reply]
import phoebe
pb = phoebe.get_passband('TESS:T')

can't work,same error

**Comment by aprsa** [comment]
I suspect that your TESS:T passband is somehow corrupt. Can you please try to re-download it:
```
import phoebe
phoebe.download_passband('TESS:T')
pb = phoebe.get_passband('TESS:T')
```
Does it work then?

**Reply by dingxu6207** [reply]
I have installed Phoebe=2.4. It is ok.
Many thanks
ding xu

**Reply by kecnry** [reply]
@dingxu6207 - did updating phoebe or re-downloading the passband do the trick?  If it required an update to phoebe to load online passbands, then that is something we definitely need to fix, so please let us know the version of phoebe you were using before (I see your original message says 2.4.3 which was the latest version until last night, which should not affect passbands at all).

Thanks!
Kyle

**Reply by dingxu6207** [reply]
The phoebe=2.4.3 only support kepler:mean and ztf:r when I used before. I have downloaded TESS:T(phoebe.download_passband('TESS:T')), It can not work. So, I try install phoebe=2.4 later. This can work.

**Comment by dingxu6207** [comment]
The phoebe=2.4.3 only support  kepler:mean and ztf:r when I used before. I  have downloaded TESS:T(phoebe.download_passband('TESS:T')), It can not work. So, I try install  phoebe=2.4 later. This can work.
